tests/main/views/test_direct_award.py

```diff
@@ -467,6 +468,98 @@ def test_end_search_redirects_to_project_page(self):
+
+    def test_award_contract_page_renders(self):
+        self.login_as_buyer()
+
+        res = self.client.get('/buyers/direct-award/g-cloud/projects/1/did-you-award-contract')
+        assert res.status_code == 200
+
+        doc = html.fromstring(res.get_data(as_text=True))
+        assert len(doc.xpath(
+            '//h1[contains(normalize-space(), "Did you award a contract for \'My procurement project\'?")]')) == 1
+        assert len(doc.xpath('//input[@type="radio"][contains(following-sibling::label, "Yes")]')) == 1
+        assert len(doc.xpath('//input[@type="radio"][contains(following-sibling::label, "No")]')) == 1
+        assert len(doc.xpath(
+            '//input[@type="radio"][contains(following-sibling::label, "We are still assessing services")]')) == 1
+        assert len(doc.xpath('//input[@type="submit"][@value="Save and continue"]')) == 1
+
+    def test_award_contract_form_action_url_is_award_contract_url(self):
+        self.login_as_buyer()
+
+        url = '/buyers/direct-award/g-cloud/projects/1/did-you-award-contract'
+
+        res = self.client.get(url)
+        assert res.status_code == 200
+
+        doc = html.fromstring(res.get_data(as_text=True))
+        assert doc.xpath(f'boolean(//form[@action="{url}"])')
+
+    def test_award_contract_error_if_no_input(self):
+        self.login_as_buyer()
+
+        res = self.client.post('/buyers/direct-award/g-cloud/projects/1/did-you-award-contract')
+
+        assert res.status_code == 400
+
+        doc = html.fromstring(res.get_data(as_text=True))
+        assert len(doc.xpath('//legend[contains(normalize-space(), "You need to answer this question.")]')) == 1
+
+    def test_award_contract_we_are_still_assessing_redirects_to_project_page(self):
+        self.login_as_buyer()
+
+        res = self.client.post('/buyers/direct-award/g-cloud/projects/1/did-you-award-contract',
+                               data={'did_you_award_a_contract': 'still-assessing'})
+
+        assert res.status_code == 302
+        assert res.location.endswith('/buyers/direct-award/g-cloud/projects/1')
+
+    def test_award_contract_yes_redirects_to_which_service_won_contract(self):
+        self.login_as_buyer()
+
+        res = self.client.post('/buyers/direct-award/g-cloud/projects/1/did-you-award-contract',
+                               data={'did_you_award_a_contract': 'yes'})
+
+        assert res.status_code == 302
+        assert res.location.endswith('/buyers/direct-award/g-cloud/projects/1/which-service-won-contract')
+
+    def test_award_contract_no_redirects_to_why_didnt_you_award_contract(self):
+        self.login_as_buyer()
+
+        res = self.client.post('/buyers/direct-award/g-cloud/projects/1/did-you-award-contract',
+                               data={'did_you_award_a_contract': 'no'})
+
+        assert res.status_code == 302
+        assert res.location.endswith('/buyers/direct-award/g-cloud/projects/1/why-didnt-you-award-contract')
```

Comment:
The above three tests (test_award_contract_we_are_still_assessing_redirects_to_project_page, test_award_contract_yes_redirects_to_which_service_won_contract, and test_award_contract_no_redirects_to_why_didnt_you_award_contract) feel like good candidates for test parametrization, which can help reduce repetition. In cases where there are just a few different variations, I think it's particularly good for aiding readability - but it is possible to create hard to parse parametrization so be careful not to overdo it. :)

You basically just add a decorater for `pytest.mark.parametrize`, then name the parameters and provide sample values. :)

Example:
```


    @pytest.mark.parametrize('form_choice, should_redirect_to_path',
                             (
                                 ('still-assessing', '/buyers/direct-award/g-cloud/projects/1'),
                                 ('yes', '/buyers/direct-award/g-cloud/projects/1/which-service-won-contract'),
                                 ('no', '/buyers/direct-award/g-cloud/projects/1/why-didnt-you-award-contract'),
                             ))
    def test_award_contract_redirects_based_on_answer(self, form_choice, should_redirect_to_path):
        self.login_as_buyer()

        res = self.client.post('/buyers/direct-award/g-cloud/projects/1/did-you-award-contract',
                               data={'did_you_award_a_contract': form_choice})

        assert res.status_code == 302
        assert res.location.endswith(should_redirect_to_path)
```

Reply:
This makes sense, I will make this change